### PR TITLE
Add rake task to run integration tests only if all unit tests pass

### PR DESCRIPTION
**Pull request by mrfishball**

### CODE REVIEW

File: lib/tasks/spec.rake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'colorize'
+
+Rake::Task[:spec].clear
+
+desc "Run all spec"
+task :spec do
+  Rake::Task['spec:controllers'].invoke
```
Comment (ericmeyer) [inline lib/tasks/spec.rake]: What is `spec:controller`?

**Review by cmvandrevala — APPROVED** [review]

**Comment by ericmeyer** [comment]
I'm a bit confused by your organization. Looking at your spec directory, it's not clear which are intended to be integration tests and which are intended to be unit tests.

Also, is there a better name than integration tests? The name is very overloaded.

**Comment by ericmeyer** [comment]
In looking at this PR, I'm thinking there was a mismatch of expectations. If this is a useful change, I can go ahead and approve, so you can merge, however it does not seem to match what I was expecting.

This was almost certainly due to us/me not communicating expectations properly.

File: lib/tasks/spec.rake (hunk 1 above)
Comment (mrfishball) [reply lib/tasks/spec.rake]: spec:controllers and spec:requests are commands for running tests for the controllers(endpoints, unit tests) and requests(integration tests). I could treat everything in the spec directory as unit tests and have a separate directory (functional) just for the functional/integration tests. would that help to clear things up?

**Review by ericmeyer — APPROVED** [review]

**Pull request merged**